src/fstab-generator/fstab-generator.c

```diff
@@ -538,7 +538,7 @@ static int parse_fstab(bool initrd) {
                          * where a symlink refers to another mount target; this works assuming the sub-mountpoint
                          * target is the final directory.
                          */
-                        r = chase_symlinks(where, initrd ? "/sysroot" : NULL,
+                        r = chase_symlinks(initrd ? (where + 8) : where, initrd ? "/sysroot" : NULL,
                                            CHASE_PREFIX_ROOT | CHASE_NONEXISTENT,
```

Comment:
Just dropping CHASE_PREFIX_ROOT appears like the much simpler and better approach?